gl.cpp

```diff
@@ -348,30 +396,35 @@ COLOR colorRGB(const GLFix r, const GLFix g, const GLFix b)
 
 GLFix nglZBufferAt(const unsigned int x, const unsigned int y)
 {
-    if(x >= SCREEN_WIDTH || y >= SCREEN_HEIGHT)
+    if (x >= SCREEN_WIDTH || y >= SCREEN_HEIGHT)
         return 0;
 
     return z_buffer[x + y * SCREEN_WIDTH];
 }
 
 //Doesn't interpolate colors even if enabled
-void nglDrawLine3D(const VERTEX *v1, const VERTEX *v2)
-{    
+void nglDrawLine3D(const VERTEX* v1, const VERTEX* v2)
+{
     //TODO: Z-Clipping
 
     VERTEX v1_p = *v1, v2_p = *v2;
     nglPerspective(&v1_p);
```

Comment:
You might run into a division by zero (or weird results) due to the missing z clipping here as well.

As a band aid, just doing

```
	if(v1_p.z < GLFix(CLIP_PLANE) || v2_p.z < GLFix(CLIP_PLANE))
		return;
```

might work.